android/app/src/main/java/app/organicmaps/search/CategoriesAdapter.java

```diff
@@ -45,14 +48,25 @@ interface CategoriesUiListener
     void onSearchCategorySelected(@Nullable String category);
   }
 
-  private CategoriesUiListener mListener;
+  private CategoriesUiListener mListener = null;
+  private boolean mIsLangSupported;
+  private Resources mEnglishResources;
 
   CategoriesAdapter(@NonNull Fragment fragment)
   {
-    if (fragment instanceof CategoriesUiListener)
-      mListener = (CategoriesUiListener) fragment;
     mResources = fragment.getResources();
     mInflater = LayoutInflater.from(fragment.requireActivity());
+
+    if (fragment instanceof CategoriesUiListener)
+    {
+      mListener = (CategoriesUiListener) fragment;
+      mIsLangSupported = DisplayedCategories.nativeIsLangSupported(Language.getDefaultLocale());
+
+      Configuration config = new Configuration(mResources.getConfiguration());
+      config.setLocale(new Locale("en"));
+      Context localizedContext = fragment.getContext().createConfigurationContext(config);
+      mEnglishResources = localizedContext.getResources();
```

Comment:
It's better to avoid initializing English resources on each startup when they are not actually used. A Bill Pugh lazy initialization pattern can be used: https://www.baeldung.com/java-bill-pugh-singleton-implementation